components/userSettings/UpdateProfilUser.tsx

```diff
@@ -44,8 +44,19 @@ export default function UpdateProfilUser({ userId, setChoices }: IProps) {
       email: d.email,
     };
 
-    if (d.newPasswordConfirm === d.newPassword) {
+    if (
+      d.newPasswordConfirm === d.newPassword &&
+      d.email.match(/[a-z0-9_\-.]+@[a-z0-9_\-.]+\.[a-z]+/i)
+    ) {
       updateUser({ variables: dataUserUpdate });
+    } else {
+      Alert.alert('Oups !', "Votre adresse mail n'est pas correct.", [
```

Comment:
```suggestion
      Alert.alert('Oups !', « Une erreur est survenur dans la mise à jour.", [
```

Pas d’indication à l’utilisateur en clair sur ce qui n’a pas marché, c’est une faille de sécurité potentielle ;)